src/main/java/org/opensearch/agent/job/AgentMonitorJob.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.agent.job;
+
+import static org.opensearch.threadpool.ThreadPool.Names.GENERIC;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.opensearch.agent.indices.IndicesHelper;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.threadpool.ThreadPool;
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * monitor agent change and trigger index summary embedding job for new agent
```

Comment:
Why need to add this job?

Reply:
`IndexSummaryEmbedding` job run with less frequency like daily, and `AgentMonitorJob` rum more frequently to run for newly created agent which has `IndexRoutingTool` configured.